lib/internal/per_context.js

```diff
@@ -1,44 +1,44 @@
+// arguments: global
```

Comment:
the important part here wasn't the global, it was creating the closure to avoid variables leaking, since this is a script. can you make a note of that here or in node.cc so we don't accidentally break things in the future?

Reply:
@devsnek I thought about that, but how does one break this from the JS side? This source is now wrapped inside a function provided by `CompileFunctionInContext`, so any declaration in this file goes into the scope of that function. Since we are in strict mode, one cannot accidentally declare a global, and as you can see here, the global object is...global.

Reply:
@joyeecheung i'm not saying the current code doesn't work, just that there should be a note mentioning that this code expects a closure of some sort. if it were changed in the future, we could end up leaking the variables like `AtomicsNotify` to all evaluations in node.js

```
> vm.runInThisContext('const a = 5')
undefined
> vm.runInThisContext('a')
5
>
```

Reply:
Maybe in node_native_module.h? Or even a README in lib?